<?xml version="1.0"?>
<launch>

    <include file="$(find evo_localizer)/launch/localizer.launch" >
        <arg name="output" value="screen" />
        <arg name="frame_fixed" value="/map" />
        <arg name="frame_odom" value="/odom" />
        <arg name="frame_robot" value="/base_footprint" />
        <arg name="frame_output" value="/pose_localizer" />
        <arg name="path_config_file" value="$(find evo_robot_sim)/config/localizer.xml" />
        <arg name="subsampling_particles_for_gui" value="10" />
        <arg name="laser_min_range_user" value="0.1" />
        <arg name="laser_max_range_user" value="9.0" />
        <arg name="top_particlecloud" value="/particlecloud" />
        <arg name="top_probmap" value="/prob_map" />
        <arg name="top_odometry" value="/odom" />
        <arg name="top_set_pose" value="/initialpose" />
        <arg name="top_global_localization" value="/clicked_point" />
        <arg name="top_static_map" value="/static_map" />
        <arg name="top_scan_1" value="/evo_robot/laser_1/scan" />
        <arg name="top_scan_2" value="/evo_robot/laser_2/scan" />
    </include>

    <node pkg="tf" type="static_transform_publisher" name="base_footprint_to_root_link" args="0 0 0 0 0 0 1 base_footprint root_link 2"/>

    <!-- Example of scan inverter usage. Currently not needed for the simulation, but used for the real roboter. -->
    <!-- <node name="scan_inverter_front" pkg="evo_scan_inverter" type="evo_scan_inverter_node" respawn="false" output="screen">
        <remap from="scan_raw" to="evo_robot/laser_1/scan" />
        <remap from="scan" to="evo_robot/laser_1/scan_raw" />
    </node> -->

    <!-- TODO: automatic argument evaluation with xacro -->
    <!-- This STP has an equivalent of the same name in the SLAM launch file. -->
    <!-- If you modify this STP, check that one as well. -->
    <arg name="b2l_x" value="0.30" />
    <arg name="b2l_z" value="-0.05" />
    <node pkg="tf" type="static_transform_publisher" name="stp_laser_1_link" args="$(arg b2l_x) 0 $(arg b2l_z) 0 0 0 1 base_link laser_1_link 2"/>

    <node name="map_server" pkg="map_server" type="map_server" args="$(find evo_robot_sim)/config/map.yaml" />

</launch>